src/PetriEngine/Reachability/ResultPrinter.cpp

```diff
@@ -214,6 +214,8 @@ namespace PetriEngine {
                 transitions.push(p.second);
             }
 
+            auto reducer = builder->getReducer();
+            
```

Comment:
You probably want to avoid shadowing the member; introduce a new variable.

Also I think 
```
reducer = reducer ? reducer : builder->getReducer();
```
Would be more sound; the reducer might already be set, and you may actually unset it.

Reply:
I updated the definition as suggested, but it should not make any change, as the reduced is set to builder if called.